src/unity/toolkits/object_detection/one_shot_object_detection/one_shot_object_detector.cpp

```diff
@@ -0,0 +1,169 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#include <algorithm>
+#include <cmath>
+#include <limits>
+#include <vector>
+#include <random>
+
+#include <boost/gil/gil_all.hpp>
+#include <boost/gil/extension/io/jpeg.hpp>
+#include <boost/gil/extension/numeric/sampler.hpp>
+#include <boost/gil/extension/numeric/resample.hpp>
+
+#include <Eigen/Core>
+#include <Eigen/Dense>
+
+#include <image/numeric_extension/perspective_projection.hpp>
+
+#include <unity/toolkits/object_detection/object_detector.hpp>
+#include <unity/toolkits/object_detection/one_shot_object_detection/one_shot_object_detector.hpp>
+
+namespace turi {
+namespace one_shot_object_detection {
+
+/* A ParameterSampler class to randomly generate different samples of parameters
+ * that can later be used to compute the transformation matrix necessary to 
+ * create image projections.
+ */
+class ParameterSampler {
+public:
+  ParameterSampler(int width, int height) {
```

Comment:
In general, we should get in the habit of using size_t instead of int when referring to sizes, counts, etc.

Reply:
Done.